### PR TITLE
Add stack-based version of Standups

### PR DESCRIPTION
**Pull request by mbrandonw**

### CODE REVIEW

File: Standups-StackBased/Standups/App.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          self.path.append(
+            .record(
+              RecordMeetingModel(standup: standup)
+            )
+          )
+        }
+        detailModel.onConfirmDeletion = { [weak detailModel, weak self] in
+          guard let self, let detailModel else { return }
+          self.standupsList.standups.remove(id: detailModel.standup.id)
+          _ = self.path.popLast()
```
Comment (stephencelis) [inline Standups-StackBased/Standups/App.swift]: Would it be safe to do this?

```suggestion
          self.path.removeLast()
```

File: Standups-StackBased/Standups/App.swift [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          self?.standupsList.standups[id: standup.id] = standup
+        }
+
+      case .meeting:
+        break
+
+      case let .record(recordModel):
+        recordModel.onDiscardMeeting = { [weak self] in
+          guard let self else { return }
+          _ = self.path.popLast()
```
Comment (stephencelis) [inline Standups-StackBased/Standups/App.swift]: Same here?

```suggestion
          self?.path.removeLast()
```

Or:

```suggestion
          _ = self?.path.popLast()
```

File: Standups-StackBased/Standups/App.swift [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            transcript: transcript
+          )
+
+          guard
+            case let .some(.detail(detailModel)) = self.path.dropLast().last
+          else {
+            return
+          }
+
+          _ = self.path.popLast()
```
Comment (stephencelis) [inline Standups-StackBased/Standups/App.swift]: Again, if it's safe:

```suggestion
          self.path.removeLast()
```

File: Standups-StackBased/Standups/StandupDetail.swift [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+  enum AlertAction {
+    case confirmDeletion
+    case continueWithoutRecording
+    case openSettings
+  }
+
+  init(
+    destination: Destination? = nil,
+    standup: Standup
+) {
```
Comment (stephencelis) [inline Standups-StackBased/Standups/StandupDetail.swift]:
```suggestion
  ) {
```

File: Standups-StackBased/Standups/App.swift (hunk 1 above)
Comment (mbrandonw) [reply Standups-StackBased/Standups/App.swift]: yep that all works, will make those changes and gonna move some of the bind stuff to separate methods.

File: Standups-StackBased/Standups/StandupDetail.swift [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        StandupDetailView(model: StandupDetailModel(standup: .mock))
+      }
+    }
+    .previewDisplayName("Happy path")
+
+    Preview(
+      message: """
+        This preview demonstrates an "unhappy path" of the application where the speech \
+        recognizer mysteriously fails after 2 seconds of recording. This gives us an opportunity \
+        to see how the application deals with this rare occurence. To see the behavior, run the \
```
Comment (stephencelis) [inline Standups-StackBased/Standups/StandupDetail.swift]:
```suggestion
        to see how the application deals with this rare occurrence. To see the behavior, run the \
```

File: Standups-StackBased/Standups/StandupsList.swift [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+   .record,
+   .detail,
+   .record,
+   .detail,
+   .record,
+    ...
+   ]
+
+   */
+
```
Comment (stephencelis) [inline Standups-StackBased/Standups/StandupsList.swift]:
```suggestion
```

**Review by stephencelis — APPROVED** [review]

**Pull request merged**